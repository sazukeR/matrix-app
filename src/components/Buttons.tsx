"use client";

interface Props {
 showRotate: boolean;
 rows: number;
 columns: number;
 handleRotate: () => void;
}

export const Buttons = ({ showRotate, rows, columns, handleRotate }: Props) => {
 return (
  <div className="buttonsContainer">
   {showRotate ? (
    <button
     disabled={rows > 0 || columns > 0 ? false : true}
     className="buttonRotate"
     onClick={handleRotate}
    >
     Girar
    </button>
   ) : (
    <button className="buttonReset" onClick={handleRotate}>
     Restablecer
    </button>
   )}
  </div>
 );
};
