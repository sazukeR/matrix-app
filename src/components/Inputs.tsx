import { ChangeEvent } from "react";

interface Props {
 showRotate: boolean;
 rows: number;
 columns: number;
 inputValue: string;
 handleRowsChange: (e: ChangeEvent<HTMLSelectElement>) => void;
 handleColumnsChange: (e: ChangeEvent<HTMLSelectElement>) => void;
 handleInputChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

export const Inputs = ({
 showRotate,
 rows,
 columns,
 inputValue,
 handleRowsChange,
 handleColumnsChange,
 handleInputChange,
}: Props) => {
 return (
  <div className="controls">
   <label className={`${!showRotate ? "orange" : ""}`}>
    Filas:
    <select
     disabled={!showRotate}
     value={rows || ""}
     onChange={handleRowsChange}
    >
     <option value="" disabled></option>
     {Array.from({ length: 10 }, (_, i) => i + 1).map((number) => (
      <option key={number} value={number}>
       {number}
      </option>
     ))}
    </select>
   </label>

   <label className={`${!showRotate ? "orange" : ""}`}>
    Columnas:
    <select
     disabled={!showRotate}
     value={columns || ""}
     onChange={handleColumnsChange}
    >
     <option value="" disabled></option>
     {Array.from({ length: 10 }, (_, i) => i + 1).map((number) => (
      <option key={number} value={number}>
       {number}
      </option>
     ))}
    </select>
   </label>

   <input
    type="text"
    value={inputValue}
    onChange={handleInputChange}
    placeholder="Añadir números"
    disabled={(rows === 0 && columns === 0) || !showRotate}
   />
  </div>
 );
};
