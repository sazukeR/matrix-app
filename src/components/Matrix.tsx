"use client";

import { useMatrix } from "@/hooks/useMatrix";
import { Inputs, Table, Buttons } from "@/components";

export const Matrix = () => {
 const {
  matrix,
  rows,
  columns,
  showRotate,
  inputValue,
  handleColumnsChange,
  handleRowsChange,
  handleInputChange,
  handleRotate,
 } = useMatrix();

 return (
  <div className="container">
   {/* <div className="controls">
    <label className={`${!showRotate ? "orange" : ""}`}>
     Filas:
     <select
      disabled={!showRotate}
      value={rows || ""}
      onChange={handleRowsChange}
     >
      <option value="" disabled></option>
      {Array.from({ length: 10 }, (_, i) => i + 1).map((number) => (
       <option key={number} value={number}>
        {number}
       </option>
      ))}
     </select>
    </label>

    <label className={`${!showRotate ? "orange" : ""}`}>
     Columnas:
     <select
      disabled={!showRotate}
      value={columns || ""}
      onChange={handleColumnsChange}
     >
      <option value="" disabled></option>
      {Array.from({ length: 10 }, (_, i) => i + 1).map((number) => (
       <option key={number} value={number}>
        {number}
       </option>
      ))}
     </select>
    </label>

    <input
     type="text"
     value={inputValue}
     onChange={handleInputChange}
     placeholder="Añadir números"
     disabled={(rows === 0 && columns === 0) || !showRotate}
    />
   </div> */}

   <Inputs
    showRotate={showRotate}
    rows={rows}
    columns={columns}
    inputValue={inputValue}
    handleRowsChange={handleRowsChange}
    handleColumnsChange={handleColumnsChange}
    handleInputChange={handleInputChange}
   />

   {/* <div className="boxTable">
    <div className="tableContainer">
     <table className="matrixTable">
      <tbody>
       {matrix.map((row, rowIndex) => (
        <tr key={rowIndex}>
         {row.map((cell, colIndex) => (
          <td key={colIndex}>{cell}</td>
         ))}
        </tr>
       ))}
      </tbody>
     </table>
     <div className={`rotate-container ${!showRotate ? "rotate" : ""}`}></div>
    </div>
   </div> */}

   <Table matrix={matrix} showRotate={showRotate} />

   {/* <div className="buttonsContainer">
    {showRotate ? (
     <button
      disabled={rows > 0 || columns > 0 ? false : true}
      className="buttonRotate"
      onClick={handleRotate}
     >
      Girar
     </button>
    ) : (
     <button className="buttonReset" onClick={handleRotate}>
      Restablecer
     </button>
    )}
   </div> */}

   <Buttons
    showRotate={showRotate}
    rows={rows}
    columns={columns}
    handleRotate={handleRotate}
   />
  </div>
 );
};
