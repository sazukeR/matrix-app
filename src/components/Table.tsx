import { TableRow } from "@/components";

interface Props {
 matrix: (string | number)[][];
 showRotate: boolean;
}

export const Table = ({ matrix, showRotate }: Props) => {
 return (
  <div className="boxTable">
   <div className="tableContainer">
    <table className="matrixTable">
     <tbody>
      {matrix.map((row, rowIndex) => (
       <TableRow key={rowIndex} row={row} />

       // <tr key={rowIndex}>
       //  {row.map((cell, colIndex) => (
       //   <td key={colIndex}>{cell}</td>
       //  ))}
       // </tr>
      ))}
     </tbody>
    </table>
    <div className={`rotate-container ${!showRotate ? "rotate" : ""}`}></div>
   </div>
  </div>
 );
};
