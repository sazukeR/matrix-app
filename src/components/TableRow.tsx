interface Props {
 row: (string | number)[];
}
export const TableRow = ({ row }: Props) => {
 return (
  <tr>
   {row.map((cell, colIndex) => (
    <td key={colIndex}>{cell}</td>
   ))}
  </tr>
 );
};
