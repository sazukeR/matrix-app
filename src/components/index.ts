export * from "./Buttons";
export * from "./Inputs";
export * from "./Matrix";
export * from "./Table";
export * from "./TableRow";
