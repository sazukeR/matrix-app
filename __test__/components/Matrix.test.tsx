import { render } from "@testing-library/react";
import { Matrix } from "@/components";

it("Should match with snapshot <Matrix />", () => {
 const { container } = render(<Matrix />);

 expect(container).toMatchSnapshot();
});
