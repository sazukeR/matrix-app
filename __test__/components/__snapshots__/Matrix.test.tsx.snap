// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match with snapshot <Matrix /> 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="controls"
    >
      <label
        class=""
      >
        Filas:
        <select>
          <option
            disabled=""
            value=""
          />
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="10"
          >
            10
          </option>
        </select>
      </label>
      <label
        class=""
      >
        Columnas:
        <select>
          <option
            disabled=""
            value=""
          />
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="10"
          >
            10
          </option>
        </select>
      </label>
      <input
        placeholder="Añadir números"
        type="text"
        value=""
      />
    </div>
    <div
      class="boxTable"
    >
      <div
        class="tableContainer"
      >
        <table
          class="matrixTable"
        >
          <tbody>
            <tr>
              <td>
                <a
                  href="/items"
                >
                   
                </a>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="rotate-container "
        />
      </div>
    </div>
    <div
      class="buttonsContainer"
    >
      <button
        class="buttonRotate"
      >
        Girar
      </button>
    </div>
  </div>
</div>
`;
